File: examples/python/upload_ai_conversation_html.py
"""
This example uses the ChatView API.
It shows how to share ChatGPT, Claude or AI assistant conversations.
Set CHATVIEW_API_TOKEN in your environment.
Never hard-code your ChatView API token.

Upload AI conversation HTML from a saved export file.
"""

import os
from pathlib import Path

from chatview import ChatViewClient


def upload_ai_conversation_html(html_path: str, plain_text_path: str | None = None) -> dict:
    client = ChatViewClient(api_token=os.environ["CHATVIEW_API_TOKEN"])
    raw_html = Path(html_path).read_text(encoding="utf-8")
    plain_text = (
        Path(plain_text_path).read_text(encoding="utf-8") if plain_text_path else None
    )
    return client.create_content(
        raw_html=raw_html,
        plain_text=plain_text,
        title="Uploaded AI conversation HTML",
        source_platform="ChatGPT",
    )


if __name__ == "__main__":
    import sys

    if len(sys.argv) < 2:
        print("Usage: upload_ai_conversation_html.py <conversation.html> [plain.txt]")
        raise SystemExit(1)
    result = upload_ai_conversation_html(sys.argv[1], sys.argv[2] if len(sys.argv) > 2 else None)
    print(result["url"])

File: examples/javascript/upload-ai-conversation-html.js
/**
 * This example uses the ChatView API.
 * It shows how to share ChatGPT, Claude or AI assistant conversations.
 * Set CHATVIEW_API_TOKEN in your environment.
 * Never hard-code your ChatView API token.
 *
 * Upload rendered AI conversation HTML captured from the browser.
 */

const CHATVIEW_API_BASE = 'https://chat-view.com/api';

const chatgptConversationHtml = document
  ? document.querySelector('.conversation')?.innerHTML
  : `<article><p>Fallback HTML when not run in a browser.</p></article>`;

async function uploadAiConversationHtml(rawHtml, plainText) {
  const token = process.env.CHATVIEW_API_TOKEN;
  if (!token) throw new Error('Set CHATVIEW_API_TOKEN.');

  const res = await fetch(`${CHATVIEW_API_BASE}/contents`, {
    method: 'POST',
    headers: {
      Authorization: `Bearer ${token}`,
      'Content-Type': 'application/json',
      Accept: 'application/json',
    },
    body: JSON.stringify({
      raw_html: rawHtml,
      plain_text: plainText,
      title: 'Uploaded AI conversation HTML',
      source_platform: 'ChatGPT',
      source_url: typeof location !== 'undefined' ? location.href : undefined,
    }),
  });

  const body = await res.json();
  if (!res.ok) throw new Error(body.message || res.statusText);
  return body;
}

if (typeof module !== 'undefined') {
  module.exports = { uploadAiConversationHtml };
}
